projects/RabbitMQ.Client/client/impl/ISession.cs

```diff
@@ -73,8 +74,7 @@ internal interface ISession
         void Close(ShutdownEventArgs reason, bool notify);
         bool HandleFrame(in InboundFrame frame);
         void Notify();
-        void Transmit(in OutgoingCommand cmd);
-        void Transmit(IList<OutgoingCommand> cmds);
-
+        void Transmit<T>(in T cmd) where T : struct, IOutgoingCommand;
+        void Transmit<T>(IList<T> cmds) where T : struct, IOutgoingCommand;
```

Comment:
Given that we never modify the commands would it make sense to use a concrete read-only structure directly and avoid any interface dispatches?

Reply:
The IList one is just used for batch publishes. So the list is populated first but we might be able to just hardcode it to use a List<T>?

Reply:
If this list is instantiated by the client, yes. If it is user-provided, I don't think so.

Reply:
Changed to concrete type since it isn't user provided.